extensions/Worker.Extensions.ServiceBus/src/ServiceBusMessageActions.cs

```diff
@@ -0,0 +1,134 @@
+            {
+                return converted;
+            }
+            // support all types listed here - https://learn.microsoft.com/en-us/dotnet/api/azure.messaging.servicebus.servicebusmessage.applicationproperties?view=azure-dotnet
+            foreach (var kvp in properties)
+            {
+                SettlementProperties settlementProperties = kvp.Value switch
+                {
+                    string stringValue => new SettlementProperties() { StringValue = stringValue },
+                    bool boolValue => new SettlementProperties() { BoolValue = boolValue },
+                    // proto does not support single byte, so use int
+                    byte byteValue => new SettlementProperties() { IntValue = byteValue },
+                    sbyte sbyteValue => new SettlementProperties() { IntValue = sbyteValue },
+                    // proto does not support short, so use int
+                    short shortValue => new SettlementProperties() { IntValue = shortValue },
+                    ushort ushortValue => new SettlementProperties() { IntValue = ushortValue },
+                    int intValue => new SettlementProperties() { IntValue = intValue },
+                    uint uintValue => new SettlementProperties() { UintValue = uintValue },
+                    long longValue => new SettlementProperties() { LongValue = longValue },
+                    // proto does not support ulong, so use double
+                    ulong ulongValue => new SettlementProperties() { DoubleValue = ulongValue },
+                    double doubleValue => new SettlementProperties() { DoubleValue = doubleValue },
+                    decimal decimalValue => new SettlementProperties() { DoubleValue = decimal.ToDouble(decimalValue) },
+                    float floatValue => new SettlementProperties() { FloatValue = floatValue },
+                    char charValue => new SettlementProperties() { StringValue = charValue.ToString() },
+                    Guid guidValue => new SettlementProperties() { StringValue = guidValue.ToString() },
+                    DateTimeOffset dateTimeOffsetValue => new SettlementProperties()
+                        { TimestampValue = Timestamp.FromDateTimeOffset(dateTimeOffsetValue) },
+                    // proto does not support DateTime, so use Timestamp from google.protobuf
+                    DateTime dateTimeValue => new SettlementProperties() { TimestampValue = Timestamp.FromDateTimeOffset(dateTimeValue) },
+                    // proto does not support Uri, so use string
+                    Uri uriValue => new SettlementProperties() { StringValue = uriValue.ToString() },
+                    // proto does not support TimeSpan, so use string
+                    TimeSpan timeSpanValue => new SettlementProperties() { StringValue = timeSpanValue.ToString() },
+                    _ => throw new NotSupportedException($"Unsupported property type {kvp.Value.GetType()}"),
```

Comment:
Is there any additional information we can provide that would help the end user understand/fix the problem with this exception?

Reply:
Added